### PR TITLE
feat: add tally/require-secret-mounts rule

### PR DESCRIPTION
**Pull request by tinovyatkin**
## Summary

- Add `tally/require-secret-mounts` rule that enforces `--mount=type=secret` on RUN instructions executing commands that require secrets — private registry credentials, API keys, cloud provider tokens, etc.
- Secrets can be mounted as **files** (`target`) or **environment variables** (`env`), mutually exclusive per command entry
- Rule is disabled by default — requires user configuration mapping command names to secret mount specs
- Auto-fix support (`--fix`, FixSafe) deterministically inserts missing `--mount=type=secret` flags

### Fix architecture

- Both `require-secret-mounts` and `prefer-package-cache-mounts` use **zero-length insertions** right after `RUN` for mount flags — they compose in a single `--fix` pass without conflicting
- `prefer-package-cache-mounts` refactored: mount additions split from script cleanup into separate targeted edits (no more full-RUN rewrites for mount-only cases)

### Configuration example

```toml
[rules.tally.require-secret-mounts]
severity = "warning"

# File-based secret
[rules.tally.require-secret-mounts.commands.pip]
id = "pipconf"
target = "/root/.config/pip/pip.conf"

# Environment-variable secret
[rules.tally.require-secret-mounts.commands.gh]
id = "gh-token"
env = "GH_TOKEN"

# AWS credentials file
[rules.tally.require-secret-mounts.commands.aws]
id = "aws"
target = "/root/.aws/credentials"
```

## Test plan

- [x] Unit tests: metadata, config validation (including target/env mutual exclusivity), 20+ check scenarios, fix-output tests
- [x] Integration lint test: fixture with mixed present/missing mounts
- [x] Integration fix tests: standalone secret mount, preserve existing cache mount, combined with `prefer-package-cache-mounts` (both mounts in single `--fix`), multi-command with cleanup (`npm install && npm cache clean --force && pip install --no-cache-dir pandas`)
- [x] Edge cases: same secret id with different targets across commands, env-based mounts
- [x] Full test suite passes (3879+ tests, 0 failures)
- [x] Schema regeneration (`make schema-gen`) produces consistent output

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added a security lint rule enforcing BuildKit secret mounts for commands that access private registries, with per-command configuration and auto-fix suggestions.

* **Documentation**
  * Added full rule docs with examples and updated supported-rules counts in README and site docs.

* **Improvements**
  * Improved fix/edit generation to preserve cache mounts, avoid overlapping edits, and relax strict fix-content assertions.

* **Tests**
  * Added extensive unit and integration tests and snapshots covering detection and fix scenarios.

### CODE REVIEW

File: internal/rules/tally/prefer_package_cache_mounts.go [hunk 1]
```diff
@@ -124,32 +122,87 @@ func (r *PreferPackageCacheMountsRule) Check(input rules.LintInput) []rules.Viol
...
+		if flags := formatRunFlags(run.FlagsUsed, existing); flags != "" {
+			tailText = flags + " " + cleanedScript
+		} else {
+			tailText = cleanedScript
+		}
+		edits = append(edits, rules.TextEdit{
+			Location: rules.NewRangeLocation(file, startLine, startCol, endLine, endCol),
+			NewText:  tailText,
+		})
+	}
```
[the remark this answers is not in the document]
Comment (tinovyatkin) [reply internal/rules/tally/prefer_package_cache_mounts.go]: ✅ Addressed in b2ddabe. Thanks @coderabbitai!

Three fixes:
- `collectNewMounts` now matches by `type+target` (not `type+target+CacheID`) so in-place mount updates are not duplicated
- Rewrite path triggers on `mountsMutated` (not just `scriptCleaned`)  
- Rewrite uses `merged` mounts (not `existing`) in `formatRunFlags` so updated sharing/id are written correctly

File: internal/rules/tally/prefer_package_cache_mounts.go [hunk 2]
```diff
@@ -98,22 +98,28 @@ func (r *PreferPackageCacheMountsRule) Check(input rules.LintInput) []rules.Viol
...
+				run:             run,
+				runLoc:          runLoc,
+				sm:              sm,
+				existing:        existing,
+				merged:          mergedMounts,
+				cleanedScript:   updatedScript,
+				scriptCleaned:   scriptCleaned,
+				cleaners:        cleaners,
+				cacheEnvEntries: cacheEnvEntries,
+			})
```
[the remark this answers is not in the document]
Comment (tinovyatkin) [reply internal/rules/tally/prefer_package_cache_mounts.go]: ✅ Addressed in 2d17720. Thanks @coderabbitai!

Added `shellVariant` to `cacheMountEditParams` and threaded the stage-specific variant through to `computeCleanupEdits`, removing the hard-coded `shell.VariantBash`.
